hierarchical_conf/hierarchical_conf.py

```diff
@@ -117,6 +117,8 @@ def _deep_update(
             if isinstance(value, Mapping) and value:
                 returned = self._deep_update(source.get(key, {}), value)
                 source[key] = returned
+            elif value is None:
+                source.pop(key, None)
```

Comment:
Would we want to choose wether the key is removed or it's persisted with the None value? There may be some situations where we would like to keep the key with the None, or am I being to picky?